### ISSUE TITLE
Extension: Closing popup during signing does not cancel signing request

### ISSUE THREAD

**Issue by jurevans** [anoma/namada-interface#1104]
We need to add handling to check if the popup for signing has been closed by the user, and treat this like a `Reject`.

Something like the following could work - we can establish a port for `signer` and check for `onDisconnect`:

```javascript
// popup.js
chrome.runtime.connect({ name: "signer" });

// background.js
chrome.runtime.onConnect.addListener(function(port) {
    if (port.name === "signer") {
        port.onDisconnect.addListener(function() {
           // Cancel the current PendingTx, send rejected event so Namadillo is aware
        });
    }
});
```

The following may also be able to catch this, so long as the background is aware of the popup tab ID:

```javascript
chrome.tabs.onRemoved.addListener(
  callback: function,
)
```
See: https://developer.chrome.com/docs/extensions/reference/api/tabs#event-onRemoved

**Comment by euharrison** [comment]
ref: https://github.com/anoma/namada-interface/pull/1101/files#r1757330875

**Comment by jurevans** [comment]
Closing, as it's fixed in #1101
